From 98b5882694c76c6ed62cfce4baae377fd61d09d7 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Tue, 04 Feb 2020 15:20:29 +0200
Subject: [PATCH] [SW-8229] reset psoc and arc on warm reset

According to new flow in spec we change default behaviour
of reset caused by PERST, we'll reset ARC and PSOC as long
as we dont have a functioning arcboot in RAM. This will
make sure we can allways deal with PERST signal.

Change-Id: I20f96948cafe93e6d8d2cbd4cf22b1377016860f
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 0503e85..8833382 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -23,6 +23,14 @@
 
 static void arcboot_rom_init(void)
 {
+	uint32_t rst_cfg;
+
+	/* on warm reset change default to reset ARC and PSOC */
+	rst_cfg = regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L);
+	rst_cfg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_ARC_MASK;
+	rst_cfg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_PSOC_MASK;
+	regmap_writel(mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L, rst_cfg);
+
 	regmap_writel(SPI_FLASH_BAUDR_DIVISOR, mmPSOC_QSPI_BAUDR);
 	regmap_writel(write_en_16MB, mmPSOC_SECURITY_SPI_WRITE_LIMIT);
 }
